File: Prototype3/Backend/models/child.py
from datetime import datetime

class Child:
    def __init__(self, id, child_name, sleep_average, treatment_id, time, informacioMedica, created_at=None):
        self.id = id
        self.child_name = child_name
        self.sleep_average = sleep_average
        self.treatment_id = treatment_id
        self.time = time
        self.informacioMedica = informacioMedica
        self.created_at = created_at if created_at else datetime.now()

    def to_dict(self):
        return {
            "id": self.id,
            "child_name": self.child_name,
            "sleep_average": self.sleep_average,
            "treatment_id": self.treatment_id,
            "time": self.time,
            "informacioMedica": self.informacioMedica,
            "created_at": self.created_at.isoformat() if self.created_at else None
        }

File: ProtypePruebas/Frontend/tapatapp_flutter/lib/core/models/child.dart
import 'package:json_annotation/json_annotation.dart';

part 'child.g.dart';

@JsonSerializable()
class Child {
  final int id;
  @JsonKey(name: 'child_name')
  final String childName;
  final int edad;
  @JsonKey(name: 'informacioMedica')
  final String informacionMedica;
  @JsonKey(name: 'tutor_id')
  final int? tutorId;
  @JsonKey(name: 'cuidador_id')
  final int? cuidadorId;
  @JsonKey(name: 'sleep_average')
  final double? sleepAverage;
  @JsonKey(name: 'treatment_id')
  final int? treatmentId;

  Child({
    required this.id,
    required this.childName,
    required this.edad,
    required this.informacionMedica,
    this.tutorId,
    this.cuidadorId,
    this.sleepAverage,
    this.treatmentId,
  });

  factory Child.fromJson(Map<String, dynamic> json) => _$ChildFromJson(json);
  Map<String, dynamic> toJson() => _$ChildToJson(this);
}